src/main/java/mate/academy/service/OrderServiceImpl.java

```diff
@@ -0,0 +1,112 @@
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+
+    @Override
+    public void placeOrder(User user, ShippingAddressRequestDto shippingAddress) {
+        Order order = createOrder(user, shippingAddress);
+        ShoppingCart shoppingCart = shoppingCartRepository.findShoppingCartByUserId(user.getId())
+                .orElseThrow(() -> new EntityNotFoundException("Can't find shopping cart "
+                        + "by user id " + user.getId()));
+        Set<OrderItem> orderItems = shoppingCart.getCartItems().stream()
+                .map(cartItem -> {
+                    OrderItem orderItem = new OrderItem();
+                    orderItem.setBook(cartItem.getBook());
+                    orderItem.setQuantity(cartItem.getQuantity());
+                    orderItem.setPrice(cartItem.getBook().getPrice());
+                    orderItem.setOrder(order);
+                    return orderItemRepository.save(orderItem);
+                })
+                .collect(Collectors.toSet());
+        shoppingCartRepository.delete(shoppingCart);
+    }
+
+    @Override
+    public List<OrderResponseDto> getAllOrders(User user,
+                                               Pageable pageable) {
+        Page<Order> allOrders = orderRepository.findAllByUserId(user.getId(), pageable);
+        return allOrders.stream()
+                .map(orderMapper::toResponseDto)
+                .toList();
+    }
+
+    @Override
+    public OrderResponseDto updateOrderStatus(User user, Long orderId,
+                                              OrderStatusDto statusDto) {
+        Order order = orderRepository.findById(orderId)
+                .orElseThrow(() -> new EntityNotFoundException("Can't find order by id "
+                        + orderId));
+        order.setStatus(statusDto.getStatus());
+        return orderMapper.toDto(orderRepository.save(order));
+    }
+
+    @Override
+    public List<OrderItemResponseDto> getAllOrderItems(User user, Long orderId, Pageable pageable) {
+        Order order = orderRepository.findByUserIdAndId(user.getId(), orderId)
+                .orElseThrow(() -> new EntityNotFoundException("Can't find order by order id "
+                        + orderId + " and user id " + user.getId()));
+        return order.getOrderItems().stream()
+                .map(orderItemMapper::toResponseDto)
+                .toList();
+    }
+
+    @Override
+    public OrderItemResponseDto getSpecificOrderItem(User user, Long orderId, Long itemId) {
+        List<OrderItemResponseDto> allOrderItems = getAllOrderItems(user,
+                orderId, Pageable.unpaged());
+        return allOrderItems.stream()
+                .filter(item -> item.getId().equals(itemId))
+                .findFirst()
+                .orElseThrow(() -> new EntityNotFoundException("Can't find order item by id "
+                        + itemId));
```

Comment:
sql side filtering is more efficient that getAll+java side filtering
make proper repository method to get single item from DB